### ISSUE TITLE
Go Code is removing a proper import even after adding it

### ISSUE THREAD

**Issue by selfup** [Microsoft/vscode-go#1266]
👋 

Here is the repo I am using (to help us all figure out what is going on here): https://github.com/selfup/go-cache-http

Current workaround:

https://github.com/selfup/go-cache-http/blob/master/main.go#L8

### Problem

**There is a UI bug**

So I have a `lib` directory at the root of my project.

I try to import it correctly: `"github.com/package-name/lib"`

And this tool just makes it disappear! Poof 💥 🚀 

I am trying to just import the functions, not force myself to do:

```golang
import (
  lib "github.com/user/package-name/lib"
)

// and then do something like
func main() {
  lib.ExportedFunction()
}
```

I would prefer (since this is an internal library to the package) to do something like:

```golang
import (
  "github.com/user/package-name/lib"
)

// and then do something like
func main() {
  ExportedFunction()
}
```

_if I am just doing something wrong please let me know_

### GIF showing what is happening

![2017-10-05 19 40 15](https://user-images.githubusercontent.com/9837366/31259599-30ac0d82-aa05-11e7-8a5b-5734d0cb8d5b.gif)

I save my work, then **_it gets rid of it_** 😂 

### Conclusion

How am I supposed to import this while using this wonderful tool? 🤔

**Unless** of course this is convention pushed by the linter then I apologize in advance for all of this!

***

If this is indeed convention, the dropdown should auto match the ending dir name (possibly).

Ex: `"github.com/user/package-name/foo"` to auto make `foo "github.com/user/package-name/foo"` if this is going to be forced from a stylistic point of view 😄

**Comment by lggomez** [comment]
I had a similar problem (in my case, adding undesired/wrong imports) and my workaround was to change the format tool to gofmt. This user setting should do the trick:

`"go.formatTool": "gofmt"`

**Comment by selfup** [comment]
Oh thanks I will try this out in a bit 😄

**Comment by ramya-rao-a** [comment]
@selfup 

I am trying to repro this. 

- Ran `go get github.com/selfup/go-cache-http`
- Opened the `go-cache-http` folder in VS Code
- Removed the `lib "github.com/selfup/go-cache-http/lib"` line from the import block in `main.go` file
- Saved the file

Instead of getting the error on missing import, the missing import got added when I saved the file.
This is because the default formatting tool `goreturns` adds missing imports and removes unused imports.

Is there anything else I should have done to get a repro?
